backend/app/Http/Controllers/ActivityController.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Activity;
+use Auth;
+
+class ActivityController extends Controller
+{
+    public function show()
+{
+    $userId = Auth::id();
+    $activities = Activity::where('user_id', $userId)->get();
+    
+    $activitiesWithTypes = [];
+    foreach ($activities as $activity) {
+        $activity->activitiable_type = $activity->activitiable_type;
+        $activitiesWithTypes[] = $activity;
+    }
```

Comment:
What is the difference of this block vs this code `$activities = Activity::where('user_id', $userId)->get();`?
I dont think you need this one,